backend/editor/entries.py

```diff
@@ -21,6 +21,91 @@ def shutdown_db():
+    else: return 'ENTRY'
+
+def create_node(label, entry, mainLanguageCode):
+    """
+    Helper function used for creating a node with given id and label
+    """
+    query = [f"""CREATE (n:{label})\n"""]
+
+    # Build all basic keys of a node
+    if (label == "ENTRY"):
+        canonicalTag = entry[3:]
+        query.append(f""" SET n.main_language = "{mainLanguageCode}" """) # Required for only an entry
+    else:
+        canonicalTag = ""
+
+    query.append(f""" SET n.id = $id """)
+    query.append(f""" SET n.tags_{mainLanguageCode} = ["{canonicalTag}"] """)
+    query.append(f""" SET n.preceding_lines = [] """)
+
+    result = session.run(" ".join(query), {"id": entry})
+    return result
+
+def add_node_to_end(label, entry):
+    """
+    Helper function which adds an existing node to end of taxonomy
+    """
+    # Delete relationship between current last node and __footer__
+    query = [f"""MATCH (a)-[r:is_before]->(b:TEXT) WHERE b.id = "__footer__" DELETE r\n"""]
+    query.append(f"""RETURN a""")
+    result = session.run(" ".join(query))
+    end_node = result.data()[0]['a']
+    end_node_label = get_label(end_node['id']) # Get current last node ID
+
+    # Rebuild relationships by inserting incoming node at the end
+    query = []
+    query.append(f"""MATCH (n:{label}) WHERE n.id = $id\n""")
+    query.append(f"""MATCH (b:{end_node_label}) WHERE b.id = $endnodeid\n""")
+    query.append(f"""MATCH (a:TEXT) WHERE a.id = "__footer__"\n""")
+    query.append(f"""CREATE (b)-[:is_before]->(n)\n""")
+    query.append(f"""CREATE (n)-[:is_before]->(a)""")
+    result = session.run(" ".join(query), {"id": entry, "endnodeid": end_node['id']})
+
+def add_node_to_beginning(label, entry):
+    """
+    Helper function which adds an existing node to beginning of taxonomy
+    """
+    # Delete relationship between current first node and __header__
+    query = [f"""MATCH (b:TEXT)-[r:is_before]->(a) WHERE b.id = "__header__" DELETE r\n"""]
+    query.append(f"""RETURN a""")
+    result = session.run(" ".join(query))
+    end_node = result.data()[0]['a']
+    end_node_label = get_label(end_node['id']) # Get current first node ID
+
+    # Rebuild relationships by inserting incoming node at the beginning
+    query = []
+    query.append(f"""MATCH (n:{label}) WHERE n.id = $id\n""")
+    query.append(f"""MATCH (b:{end_node_label}) WHERE b.id = $startnodeid\n""")
+    query.append(f"""MATCH (a:TEXT) WHERE a.id = "__header__"\n""")
+    query.append(f"""CREATE (n)-[:is_before]->(b)\n""")
+    query.append(f"""CREATE (a)-[:is_before]->(n)""")
+    result = session.run(" ".join(query), {"id": entry, "startnodeid": end_node['id']})
+
+def delete_node(label, entry):
+    """
+    Helper function used for deleting a node with given id and label
+    """
+    # Finding node to be deleted using node ID
+    query = []
+    query.append(f"""MATCH (n:{label})-[:is_before]->(a) WHERE n.id = $id\n""")

```

Comment:
Here also renaming `a` and `b` to `next_node` and `previous_node` would help :-)